src/components/CandidateAvatar/CandidateAvatar.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from '../../packages/bem';
+import styles from './CandidateAvatar.scss';
+
+const { block, elem } = bem({
+    name: 'CandidateAvatar',
+    classnames: styles,
+    propsToMods: ['context']
+});
+
+const context = percentage => {
+    if (percentage <= 33) {
+        return 'bad';
+    }
+    if (percentage <= 66) {
+        return 'warning';
+    }
+    return 'good';
+};
+
+const CandidateAvatar = props => {
+    const { imageUrl, matchPercentage, showPercentageOnHover, size, ...rest } = props;
+
+    const fixedSize = Math.round(Math.max(32, size));
+    const radius = fixedSize / 2 - 2;
+    const percentage = Math.max(0, Math.min(100, matchPercentage));
+    const circumference = 2 * radius * Math.PI;
+    const strokeDasharray = `${(percentage * circumference) / 100} 999`;
+
+    return (
+        <div
+            {...rest}
+            {...block(props)}
+            style={{
+                width: fixedSize,
+                height: fixedSize
+            }}
+        >
+            <div
+                {...elem('image', props)}
+                style={{
+                    backgroundImage: `url(${imageUrl})`
+                }}
+            >
+                {!!showPercentageOnHover && (
+                    <div {...elem('percentage', props)}>{`${percentage}%`}</div>
+                )}
+            </div>
+            <svg
+                {...elem('ring', props)}
+                style={{
+                    width: fixedSize,
+                    height: fixedSize
+                }}
+            >
+                <circle
+                    r={radius}
+                    cx={fixedSize / 2}
+                    cy={fixedSize / 2}
+                    strokeWidth="4"
+                    {...elem('circle', {
+                        ...props,
+                        context: context(percentage)
```

Comment:
@antipin when looking at my console log I see the following warning: `BEM component "CandidateAvatar" prabably has incorrect declaration of "context"`. Doesn't this make sense? Do you see another way to compute the context on the basis of `matchPercentage`? Also, the warning has a typo ("prabably") ;)

Reply:
@antipin

Reply:
Hi there, I’ve investigated the issue.
In this case we have `context` modifier which doesn't depend neither of props nor of state.
It's kind of a new (but absolutely sensible) use-case for BEM library, which was not addressed so far. Good thing is that it works, Warning is just a warning and it is onited in production mode. So far just ignore it.

I created a ticket to address it https://trello.com/c/e5cNklRv/60-address-a-way-of-creating-modifiers-that-do-not-depend-neither-of-props-nor-of-state